src/shared/sol-util.h

```diff
@@ -159,3 +168,60 @@ align_power2(unsigned int u)
  *       elements found are returned.
  */
 struct sol_vector sol_util_str_split(const struct sol_str_slice slice, const char *delim, size_t maxsplit);
+
+/* Sets the F_SETFL flags and sets FD_CLOEXEC in fd.
+ * If either operation fails, closes the fd and returns -1.
+*/
+static inline int fd_set_flags_cloexec(int fd, int flags)
```

Comment:
Maybe there should be a `sol_util_posix_fd_set_flags()` function instead? There are many other places where `fcntl(F_GETFD)` / `fcntl(F_SETFD)` are used, each with their own copy-pasted error handling.